search/migrations/0012_insert_profile_2020.py

```diff
@@ -0,0 +1,94 @@
+# Generated by Django 3.0.7 on 2020-08-23 10:17
+
+from django.db import migrations
+import csv
+import re
+
+from psycopg2.extras import NumericRange
+
+from search.models import SubjectName, LanguageName
+
+
+def load(apps, schema_editor):
+    School = apps.get_model("search", "School")
+    HighSchoolClass = apps.get_model("search", "HighSchoolClass")
+    ExtendedSubject = apps.get_model("search", "ExtendedSubject")
+    Language = apps.get_model("search", "Language")
+
+    with open("csvs/publiczne_oddzialy_2020.csv", newline="") as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=";")
+        for n, row in enumerate(csv_reader):
+            if n > 0:
+                regon = row[7].zfill(9)
+                school = School.objects.filter(public_institution_data__regon=regon)
+                if (
+                    not school.exists()
+                    or school.get().school_type != "liceum ogólnokształcące"
+                ):
+                    continue
+                school = school.get()
+                class_name = row[1]
+                type = {
+                    "ogólnodostępny": "O",
+                    "mistrzostwa sportowego": "MS",
+                    "dwujęzyczny": "D",
+                    "międzynarodowy": "M",
+                    "wstępny": "DW",
+                    "sportowy": "S",
+                    "integracyjny cz. ogólnodostępna": "I-o",
+                    "integracyjny cz. dla kandydatów z orzeczeniem o potrzebie kształcenia specjalnego": "I-i",
+                    "przygotowania wojskowego": "PW",
+                }.get(row[35], "")
+                hsc = HighSchoolClass(
+                    type=type,
+                    name=class_name,
+                    school=school,
+                    year=NumericRange(2020, 2022, "[)"),
```

Comment:
2021?

Reply:
it’s right-open interval. 
```Regardless of the bounds specified when saving the data, PostgreSQL always returns a range in a canonical form that includes the lower bound and excludes the upper bound, that is [).```
https://docs.djangoproject.com/en/3.1/ref/contrib/postgres/fields/#django.contrib.postgres.fields.IntegerRangeField

Reply:
This field shouldn’t be a range in the first place, but blame young, silly me. Someone should migrate it in the future